src/platform/packages/shared/shared-ux/error_boundary/src/ui/section_error_boundary.tsx

```diff
@@ -63,6 +65,7 @@ class SectionErrorBoundaryInternal extends React.Component<
   }
 
   componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
+    apm.captureError(Object.assign({}, error, { context: REACT_FATAL_ERROR_EVENT_TYPE }));
```

Comment:
This would end up something like `context.custom.context` in the APM error document, is that what you are expecting the format to be? Should we modify it to be `react_error_type`

Reply:
Between `context.custom.context` and `context.custom.react_error_type`, the latter makes more sense. Thanks, I will update it!

Reply:
Tim can you mutate the error instead of spreading it onto a new object? It changes the prototype and weird stuff will happen

Reply:
@tsullivan @vigneshshanmugam do we have a type/category label in Kibana? Without being able to search or filter/group based on this field, I don't see much use for adding it.

Reply:
@maryam-saeidi you can:
- add a runtime mapping to a data view (this only works in discover)
- make context.custom searchable by editing the logs-apm@custom index template (https://www.elastic.co/guide/en/observability/current/apm-custom-index-template.html) 
- move context.custom.react_error_type to labels.react_error_type in the logs-apm.error@custom pipeline (https://www.elastic.co/guide/en/observability/current/apm-ingest-pipelines.html)

I would probably recommend the latter. @vigneshshanmugam maybe also a nice feature request for the RUM agent? Does the node.js agent support custom labels on error events?

Reply:
@dgieselaar Thanks for providing the possible options <3

I would vote for moving the labels as well if there is no better option.

Btw, based on your [comment](https://github.com/elastic/observability-dev/issues/4222#issuecomment-2642783030), I implemented a custom error like this:
```
class StartOrEndError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'StartOrEndError';
  }
}
```
And now the error is reported in RUM as:
![image](https://github.com/user-attachments/assets/823cc9f0-6348-447b-841b-877d0ba17f41)

And is searchable using `error.exception.type:"StartOrEndError"`, so what if we throw a custom `FatalReactError` in this case? Is there a downside to it?

Reply:
It means you lose the exception type (and the stack trace) of the original error

Reply:
Can we somehow keep the original error in context?

Reply:
well it would essentially be a nested document, which you probably want to avoid. your goal is to be able to filter on whether it's caught by the error boundary?

Reply:
> Does the node.js agent support custom labels on error events?

Node.js agent supports adding [labels](https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#apm-add-labels) to current transaction and also supports global labels similar to the RUM agent which would add the labels to all events. There is no event specific API on the rum agent side for now.

Reply:
@maryam-saeidi it will group by FatalReactError then so you cannot group anymore easily by the underlying error type. What are you trying to achieve by grouping on FatalReactError?

Reply:
Better visibility of errors in our APM view. Let's move forward with this approach, and we can always change it if we see an actual issue. Having this code in production and experimenting with the result would be more valuable than having a long discussion here.

Reply:
What is "this approach"? The current state of the PR?

Reply:
This one: [comment](https://github.com/elastic/kibana/pull/209006#discussion_r1948912084)

Reply:
Hm ok, I'm not sure if I get why the place where an error is captured is more important than what error it is, but we can change later I guess.